### DISCUSSION TITLE
Access gpt4all-ui on local network

### DISCUSSION THREAD

**Discussion by maddes8cht** [ParisNeo/lollms-webui#175] [Q&A]
After startup I can access the gpt4all web interface on localhost.9600. 
Is there a way to make the web interface accessible on the local network?

**Comment by ParisNeo** [comment]
Yes
Replace localhost by 0.0.0.0 that makes it visible anywhere. I even setup a server at home and used my phone to test the ui.

**Reply by gilbrotheraway** [reply]
hello,

first of all great app it shows you have an inquiring mind
i have the same intent of running this on a server but it's headless and i don't have a monitor, is there a way to change the localhost in the files?
also is the lollms repo being updated? 

edit: does the ollms repo use and how can i point it to exllamav2?

Thank you very much.

**Comment by Alexandersss** [comment]
Hi In which file should I make the change exactly?
I also tried the command ./webui.sh --host 0.0.0.0. but don't work..

**Comment by ghost** [comment]
Didn't work for me.  Still not accessible via local network :(

**Comment by maddes8cht** [comment]
In
GPT4All\configs\local_default.yaml host : 0.0.0.0
but in browser of course you need the ip or machine name of the host
so its
hostip:9600

**Reply by ghost** [reply]
Since im using the newer UI its not under GPT4all, but i did update the yaml in the config folder. Oh and its not a FW issue, i have another web process running on the same box. ( no GUI.. i run everything remote to it .. )

Oh, and not complaining, i ran it on my desktop to test, seems like a wonderful thing.   But gotta run it on my server or its dreadfully slow :)

**Reply by ghost** [reply]
i was able to pass in parameters when i called app.py .. so im good.

**Reply by cgiakmoz** [reply]
Hi.  Looks like a great API.
But is there yet a solution as to WHICH *.py files or config files need to be changed (as of v.6 of LoLLMs).

Have installed on an Ubuntu Headless Server with ./linux_run.sh, but need to have external / across LAN / remote access by browser (I.e. not [http://localhost:9600]) ?

**Reply by ghost** [reply]
if you are going to run it as a server, just pass the parameters when you call app.py from whatever mechanism you are using.  Dont really need to edit anything.

**Reply by cgiakmoz** [reply]
Hi Nurb432. Thanks for replying.
I saw your answer on 23-Jun-2023 of 'pass the parameters when called app.py'.  But I am not normally a python user. Cannot work out exactly what you mean. 

On my LAN Headless (UBUNTU) server, I tried the following - 
- Having found an app.py file in the webui folder.  Using text editor NANO app.ppy ..., 
- I searched for words "localhost" and "host" and tried changing these instances of "localhost" changed to "192.168.0.123"  
- ... and also later tried changing to "0.0.0.0" ... 

Unfortunately, the browser on GUI workstation still cant connect to 192.168.0.123 ... says "connection refused"

**Reply by ghost** [reply]
id have to reinstall it all ( i gave up on it for other reasons ) for the exact parameters now but the idea is  my service would have done  " python - path to -app.py --host 0.0.0.0 "  ( there is one to change port too )   Instead of calling any .sh file they might have distributed with it, i just did it via the app.py file directly.             if i was in the folder to do it manually id just start the virtual env then type "python ./app.py --host 0.0.0.0"  ( no quotes )     i did NOT edit app.py since it takes command line parameters.

**Reply by cgiakmoz** [reply]
Hi Nurb, Thanks a lot for the help.
I managed to get the interface working across the lan by using the following command:

$>   cd ~/localai/lollms-webui/lollms-webui/scripts/linux
$>  ./linux_run.sh --host 0.0.0.0

As the run command completes, it then gives a localhost http, as well as a LAN address like http://192.168.7.55

All's good!

**Reply by ghost** [reply]
yay :)    Wasn't sure it would work with the supplied sh file, but i knew it did with the raw .py app when i messed with it.   But either way, it works, that is what matters.   I have not looked at the shell file, but i assume it handles the virtual environment stuff for you. I do that manually.

**Comment by ParisNeo** [comment]
Hi and sorry for being late. i was too buzy lately and didn't have time to check on issues.
Well, now if you want to use a server, I advise you tto use lollms as backend server and select lollms remote nodes as binding in the webui. With this, you protect your data that stays on your own machine and each user will have its own database.
Since the ui has no authentication mechanism, if many people on your network use the tool they'll be seeing each other's messages.
You can watch my last video if you want to learn how to do that:
https://youtu.be/MxXNGv1zJ1A

**Reply by ghost** [reply]
This is just for me, and i do have auth ( and ssl ) setup on my nginx proxy if i want to publish outside.  But if i ever need to support multi user, tks for the suggestion.
